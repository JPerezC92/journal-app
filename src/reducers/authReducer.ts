import { createSlice, PayloadAction } from "@reduxjs/toolkit";

export interface AuthState {
  isLoggedIn: boolean;
  displayName: string;
  uuid: string;
}

const initialState: AuthState = {
  isLoggedIn: false,
} as AuthState;

export const authSlice = createSlice({
  name: "[Auth]",
  initialState,
  reducers: {
    login: (_, action: PayloadAction<AuthState>): AuthState => action.payload,
    logout: () => initialState,
  },
});

export const { login, logout } = authSlice.actions;

const authReducer = authSlice.reducer;

export default authReducer;
