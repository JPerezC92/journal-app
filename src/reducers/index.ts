export * from "./authReducer";
export * from "./uiReducer";
export * from "./notesReducer";
