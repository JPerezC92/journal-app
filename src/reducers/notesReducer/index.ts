export * from "./notesReducer";
export * from "./notesThunks";
