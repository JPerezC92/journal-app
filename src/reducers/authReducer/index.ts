import authReducer from "./authReducer";

export * from "./authReducer";
export * from "./authReducerTypes";
export * from "./authThunks";

export default authReducer;
