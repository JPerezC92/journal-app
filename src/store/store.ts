import { configureStore } from "@reduxjs/toolkit";
import { uiReducer, authReducer, notesReducer } from "../reducers";

export const reducers = {
  authReducer,
  uiReducer,
  notesReducer,
};

export const store = configureStore({
  reducer: reducers,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
