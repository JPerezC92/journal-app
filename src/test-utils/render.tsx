import { configureStore } from "@reduxjs/toolkit";
import { render, RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { uiReducer, authReducer, notesReducer } from "../reducers";

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import("/home/philip/projects/journal-app/node_modules/@testing-library/dom/types/queries"),
    HTMLElement
  >
) => {
  const store = configureStore({
    reducer: {
      authReducer,
      uiReducer,
      notesReducer,
    },
  });
  const reduxStore: React.FC = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return { component: render(ui, { wrapper: reduxStore, ...options }), store };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
