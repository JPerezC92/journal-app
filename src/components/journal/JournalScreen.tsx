import { useSelector } from "react-redux";
import { RootState } from "../../store/store";
import NoteScreen from "../notes/NoteScreen";
import NothingSelected from "./NothingSelected";
import Sidebar from "./Sidebar";

const JournalScreen = () => {
  const { active } = useSelector((state: RootState) => state.notesReducer);
  return (
    <div className="journal__main-content animate__animated animate__fadeIn animate__faster">
      <Sidebar />

      <main>{active ? <NoteScreen /> : <NothingSelected />}</main>
    </div>
  );
};

export default JournalScreen;
