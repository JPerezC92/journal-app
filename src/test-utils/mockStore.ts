import {
  AnyAction,
  CombinedState,
  combineReducers,
  configureStore,
  createSlice,
} from "@reduxjs/toolkit";
import { reducers } from "../store/store";

const storeCleanSlice = createSlice({
  name: "store",
  initialState: {},
  reducers: {
    clean: () => {},
  },
});

export const storeCleanActions = storeCleanSlice.actions;
const storeCleanReducer = storeCleanSlice.reducer;
const combinedReducers = combineReducers({ ...reducers, storeCleanReducer });

const testReducers = (
  state: CombinedState<any> | undefined,
  action: AnyAction
) => {
  if (action.type === "store/clean") {
    state = undefined;
  }
  return combinedReducers(state, action);
};

export const mockStore = configureStore({
  reducer: testReducers,
});

export type MockRootState = ReturnType<typeof mockStore.getState>;
export type MockAppDispatch = typeof mockStore.dispatch;
