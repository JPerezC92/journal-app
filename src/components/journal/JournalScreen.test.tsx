import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { notesActions } from "../../reducers";
import { store } from "../../store/store";
import { mockStore } from "../../test-utils/mockStore";
import JournalScreen from "./JournalScreen";

describe("Test on <JournalScreen />", () => {
  test("should match the snapshot correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <JournalScreen />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("should render <NothingSelected />", async () => {
    const component = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <JournalScreen />
        </MemoryRouter>
      </Provider>
    );

    const aside = component.getByRole("complementary");

    expect(aside).toBeInTheDocument();
    expect(component.container).toHaveTextContent("Select something");
    expect(component.container).toHaveTextContent("or create an entry");
  });

  test("should render <NoteScreen />", async () => {
    const component = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <JournalScreen />
        </MemoryRouter>
      </Provider>
    );

    mockStore.dispatch(
      notesActions.setNoteActive({
        id: "1",
        title: "Test title",
        body: "Test body",
        date: 1626208415067,
        imageUrl:
          "https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png",
      })
    );
    const titleField = component.getByPlaceholderText("Some awesome title");
    const img = component.getByRole("img");
    const aside = component.getByRole("complementary");

    expect(aside).toBeInTheDocument();
    expect(titleField).toHaveValue("Test title");
    expect(component.container).toHaveTextContent("Test body");
    expect(img).toHaveAttribute(
      "src",
      "https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png"
    );
  });
});
