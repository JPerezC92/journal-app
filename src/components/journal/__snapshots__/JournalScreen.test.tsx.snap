// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <JournalScreen /> should match the snapshot correctly 1`] = `
<div
  className="journal__main-content animate__animated animate__fadeIn animate__faster"
>
  <aside
    className="journal__sidebar"
  >
    <div
      className="journal__sidebar-navbar mt-1"
    >
      <h3>
        <i
          className="far fa-moon"
        >
           
        </i>
        <span>
           
        </span>
      </h3>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </div>
    <div
      aria-label="button"
      className="journal__new-entry"
      onClick={[Function]}
    >
      <i
        className="far fa-calendar-plus fa-5x"
      />
      <p
        className="mt-5"
      >
        New entry
      </p>
    </div>
    <div
      className="journal__entries"
    />
  </aside>
  <main>
    <div
      className="nothing__main-content"
    >
      <p>
        Select something
        <br />
        or create an entry
      </p>
      <i
        className="far fa-star fa-4x mt-5"
      />
    </div>
  </main>
</div>
`;
